plugins/bookshortcuts.koplugin/main.lua

```diff
@@ -0,0 +1,111 @@
+local ConfirmBox = require("ui/widget/confirmbox")
+local DataStorage = require("datastorage")
+local Dispatcher = require("dispatcher")
+local FFIUtil = require("ffi/util")
+local LuaSettings = require("luasettings")
+local PathChooser = require("ui/widget/pathchooser")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local util = require("util")
+local _ = require("gettext")
+
+local BookShortcuts = WidgetContainer:new{
+    name = "bookshortcuts",
+    shortcuts = LuaSettings:open(DataStorage:getSettingsDir() .. "/bookshortcuts.lua"),
+    updated = false,
+}
+
+function BookShortcuts:onDispatcherRegisterActions()
+    for k,v in pairs(self.shortcuts.data) do
+        if util.fileExists(k) then
+            local directory, filename = util.splitFilePathName(k) -- luacheck: no unused
+            Dispatcher:registerAction(k, {category="none", event="BookShortcut", title=filename, filemanager=true, arg=k,})
+        end
+    end
+end
+
+function BookShortcuts:onBookShortcut(path)
+    if util.fileExists(path) then
+        local ReaderUI = require("apps/reader/readerui")
+        ReaderUI:showReader(path)
```

Comment:
We have a `ReaderUI:switchDocument(new_file)` that could work from when already in Reader.

Reply:
`showReader` works in both the reader and FM so simpler

Reply:
Does it properly close the current document (flush settings, save highlights into book, survive a prompt for password protected books, etc...)? Looks like it does not :onClose() (but may be something else does it) ?

Reply:
Haven't had an isue with it so far. it will eventually call `doShowReader` with https://github.com/koreader/koreader/blob/1dc8f8cda9293736961755047baf2af57e5517b4/frontend/apps/reader/readerui.lua#L599-L601

but as the note says:
https://github.com/koreader/koreader/blob/1dc8f8cda9293736961755047baf2af57e5517b4/frontend/apps/reader/readerui.lua#L522-L525

Reply:
The only advantage `switchDocument` is left with are the weird `Close*Menu` events, which are... kind of odd ducks. (I don't remember all the details, but IIRC, it was marginal use-cases and very weird/old design ^^).